Stale-bot: Do not mark issues stale when pending maintainer action

yongtang opened the pull request:
If an issue has been labeled as 'answered' or 'needs info', or a PR has been labeled with 'needs update' by maintainers,
then it will be marked with stale after 30 days and +7 days to close.

Any activity in between will unstale, and 'answered,needs info' in issues and 'needs update' in PRs will be removed.

So authors can exempt the issue or PR from stale as long as there is some activity.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Increase stale's threshold to 90+14 days

We originally set the stale's threshold as 30+7 days.

However, after observe some time it looks like 30+7 days
might not fit into the traffic flow of coredns, as some
reviews (or issues) may have a long turnaround time.

This PR increase the threshold to 90+14days to better
fit coredns' traffic activity.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

chrisohaver reviewed (changes requested):
I dunno. I think 30+7 days is fine. I think it will keep things moving.   For the most part it will act like a monthly reminder to maintainers of open PRs awaiting review. A reminder once every 90 days wouldn’t be as effective.

yongtang committed:
Update PR with 30+7 policy, and issue 60+15 policy.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

chrisohaver reviewed (changes requested):
I think the initial 30+7 numbers are fine for now.
If after a couple of months we find that notifications for idle issues/prs every 30 days is too frequent, we can increase them. But I don't think we can draw that conclusion after a just a week.

The already implemented exception is enhancements, which will frequently sit idle a long time before someone offers to implement them, and then idle a long time before a PR is opened.

Bugs/questions, and PRs are more urgent, and should get our more immediate attention.  Being reminded once every 30 days that these are lingering is fine IMO.  Longer than that kind of defeats the purpose.  IOW, it's not desirable for a question or bug or PR to sit idle for more than 30 days.

yongtang committed:
Revert "Update PR with 30+7 policy, and issue 60+15 policy."

This reverts commit 29bb824103bb0c5a50d31dff57ce6dcce4c83352.

Revert "Increase stale's threshold to 90+14 days"

This reverts commit 4ba30211ce3baee4371460b73ddc37850aa9fa7e.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Only adds stale with labels 'answered,needs info' in issues and 'needs update' in PRs

If an issue has been labeled as  'answered' or 'needs info',
or a PR has been labeled with 'needs update' by maintainers,
then it will be marked with stale after 30 days and +7 days to close.

Any activity in between will unstale, and 'answered,needs info' in issues and 'needs update'
in PRs will be removed: Author can exempt the issue or PR from stale as long as
there is some activity.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

chrisohaver reviewed (approved):
LGTM!
